unpackaged/vesuvio_calibration/calibrate_vesuvio_uranium_martyn_MK5.py

```diff
@@ -379,114 +379,155 @@ def _normalise_sample_by_background(self):
 #----------------------------------------------------------------------------------------
 
   def _fit_bragg_peaks(self):
-    #ESTIMATE PEAK POSITIONS USING D SPACING
-    peak_positions = self._estimate_bragg_peak_positions()
-    num_peaks, num_spectra = peak_positions.shape
+    estimated_peak_positions_all_spec = self._estimate_bragg_peak_positions()
+    num_estimated_peaks, num_spectra = estimated_peak_positions_all_spec.shape
+
     self._prog_reporter = Progress(self, 0.0, 1.0, num_spectra)
 
-    #CREATE OUTPUT PARAMETERS TABLE
-    peaks_table = self._output_workspace_name + '_Peak_Parameters'
-    CreateEmptyTableWorkspace(OutputWorkspace=peaks_table)
+    output_parameters_tbl_name = self._output_workspace_name + '_Peak_Parameters'
+    self._create_output_parameters_table_ws(output_parameters_tbl_name, num_estimated_peaks)
 
-    #PARAM NAMES FOR LINEAR BACKGROUND
-    param_names = ['f0.A0', 'f0.A1']
+    output_workspaces = []
+    for index, estimated_peak_positions in enumerate(estimated_peak_positions_all_spec.transpose()):
+        spec_number = self._spec_list[0]+index
+        self._prog_reporter.report("Fitting to spectrum %d" % spec_number)
 
-    #ADD OTHER PARAM NAMES DEPENDING UPON CHOSEN PEAK FUNCTION
-    for i in range(num_peaks):
-      param_names += ['f' + str(i) + '.' + name for name in self._func_param_names.values()]
+        find_peaks_output_name = self._sample + '_peaks_table_%d' % spec_number
+        fit_peaks_output_name = self._output_workspace_name + '_Spec_%d' % spec_number
+        fit_results = self._fit_peaks_to_spectra(index, spec_number, estimated_peak_positions, find_peaks_output_name,
+                                                 fit_peaks_output_name)
 
-    #GENERATE ERROR NAMES FROM PARAM NAMES, ZIP TO GET COL NAMES
-    err_names = [name + '_Err' for name in param_names]
-    col_names = [element for tupl in zip(param_names, err_names) for element in tupl]
+        fit_results_u = None
+        if not fit_results['status'] == "success":
+            self._prog_reporter.report("Fitting to spectrum %d without constraining parameters" % spec_number)
+            fit_results_u = self._fit_peaks_to_spectra(index, spec_number, estimated_peak_positions, find_peaks_output_name,
```

Comment:
This is just a question for my own understanding, what does the `_u` denote at the end of this variable?

Reply:
It stands for `unconstrained`, I don't see any reason not to change the variable name to make this clearer.